File: 2966-divide-array-into-arrays-with-max-difference/2966-divide-array-into-arrays-with-max-difference.cpp
class Solution {
public:
    vector<vector<int>> divideArray(vector<int>& nums, int k) {
        vector<vector<int>>ans;
        int n=nums.size();
        sort(nums.begin(),nums.end());
        for(int i=0;i<n;i+=3){
            int fn=nums[i],sn=nums[i+1],tn=nums[i+2];
            if(sn-fn<=k && tn-sn<=k && tn-fn<=k){
                ans.push_back({fn,sn,tn});
            }else{
                ans.clear();
                break;
            }
        }
        return ans;
    }
};

File: 2966-divide-array-into-arrays-with-max-difference/2966-divide-array-into-arrays-with-max-difference.java
class Solution {
    public int[][] divideArray(int[] nums, int k) {
        int n=nums.length;
        int[][] ans = new int[n/3][3];
        int[][]eans=ans;
        int i=0;
        Arrays.sort(nums);
        while(i<n){
            int val1=nums[i],val2=nums[i+1],val3=nums[i+2];
            if(val2-val1>k || val3-val2>k || val3-val1>k){
               return new int[0][];
            }
            int[] curr=new int[3];
            curr[0]=val1;curr[1]=val2;curr[2]=val3;
            ans[i/3]=curr;
            i+=3;
        }
        return ans;
    }
}